src/components/nodes/boundaryEvent/boundaryEvent.vue

```diff
@@ -125,6 +125,11 @@ export default {
     resetToInitialPosition() {
       this.shape.position(this.validPosition.x, this.validPosition.y);
       this.allowSetNodePosition = true;
+
+      if (this.invalidTargetElement) {
+        resetShapeColor(this.invalidTargetElement);
+        this.invalidTargetElement = null;
+      }
```

Comment:
This code ensures that any previously invalid element has it's colour reset when the boundary event returns to its initial position.

When working with pools, the pool colour wouldn't reset from red to the default pool colour because the logic for handling the colour change didn't detect that the target of the boundary event has changed, as in every case it appeared to be over the pool.